src/py/flwr/client/mod/localdp_mod.py

```diff
@@ -0,0 +1,129 @@
+from flwr.common.constant import MessageType
+from flwr.common.context import Context
+from flwr.common.differential_privacy import (
+    add_localdp_gaussian_noise_to_params,
+    compute_clip_model_update,
+)
+from flwr.common.message import Message
+
+
+class LocalDpMod:
+    """Modifier for local differential privacy.
+
+    This mod clips the client model updates and
+    add noise to the params before sending them to the server.
+
+    It operates on messages with type MessageType.TRAIN.
+
+    Parameters
+    ----------
+    clipping_norm : float
+        The value of the clipping norm.
+    sensitivity : float
+        The sensitivity of the client model.
+    epsilon : float
+        The privacy budget.
+        Smaller value of epsilon indicates a higher level of privacy protection.
+    delta : float
+        The failure probability.
+        The probability that the privacy mechanism
+        fails to provide the desired level of privacy.
+        A smaller value of delta indicates a stricter privacy guarantee.
+
+    Examples
+    --------
+    Create an instance of the local DP mod and add it to the client-side mods:
+
+    >>> local_dp_mod = LocalDpMod( ... )
+    >>> app = fl.client.ClientApp(
+    >>>     client_fn=client_fn, mods=[local_dp_mod]
+    >>> )
+    """
+
+    def __init__(
+        self, clipping_norm: float, sensitivity: float, epsilon: float, delta: float
+    ) -> None:
+        if clipping_norm <= 0:
+            raise ValueError("The clipping norm should be a positive value.")
+
+        if sensitivity < 0:
+            raise ValueError("The sensitivity should be a non-negative value.")
+
+        if epsilon < 0:
+            raise ValueError("The epsilon should be a non-negative value.")
+
+        if delta < 0:
+            raise ValueError("The delta should be a non-negative value.")
+
+        self.clipping_norm = clipping_norm
+        self.sensitivity = sensitivity
+        self.epsilon = epsilon
+        self.delta = delta
+
+    def __call__(
+        self, msg: Message, ctxt: Context, call_next: ClientAppCallable
+    ) -> Message:
+        """Perform local DP on the client model parameters.
+
+        Parameters
+        ----------
+        msg : Message
+            The message received from the server.
+        ctxt : Context
+            The context of the client.
+        call_next : ClientAppCallable
+            The callable to call the next middleware in the chain.
+
+        Returns
+        -------
+        Message
+            The modified message to be sent back to the server.
+        """
+        if msg.metadata.message_type != MessageType.TRAIN:
+            return call_next(msg, ctxt)
+
+        fit_ins = compat.recordset_to_fitins(msg.content, keep_input=True)
+        server_to_client_params = parameters_to_ndarrays(fit_ins.parameters)
+
+        # Call inner app
+        out_msg = call_next(msg, ctxt)
+        fit_res = compat.recordset_to_fitres(out_msg.content, keep_input=True)
```

Comment:
In between those lines, we need to check whether `out_msg.has_content()`:
```suggestion
        out_msg = call_next(msg, ctxt)
        fit_res = compat.recordset_to_fitres(out_msg.content, keep_input=True)
```